File: practical3/trapezodial.f90
program Trap
  implicit none


  integer::N=12, i
  real(kind=4):: a=0.0, b_deg=60.0
  real(kind=4):: pi, b_rad, area, mult_rad


  ! Convert pi/3 to radians 
  pi=atan(1.0)*4.0
  b_rad=(pi*b_deg)/180.0
  write(6, *) 'The value of pi/3 in radians', b_rad

  
  ! f(x0) + f(xN)
  area = tan(a)+tan(b_rad)
  write(6, *) 'Initial area', area


  ! 2*f(x1) + ... + 2*f(xN-1)
  do i=5, 55, 5
     area=area+2*tan((pi*i)/180.0)
     write(6, *) 'New area at x', i/5, '= ' , area
  end do


  ! Multiply with b-a/2N
  mult_rad=pi*((b_deg-a)/(2*N))/180.0
  area = mult_rad * area


  ! Approximated result 
  write(6, *) 'Trapezoid result ', area
  ! Real Result
  write(6, *) 'Real result ', log(2.0)


  stop
end program

File: practical3/trapezodial.c
#include<stdio.h>
#include<math.h>


int main() {

  int N =12, i;
  float a =0.0, b_deg=60.0;

  //convert pi/3 to radians
  float b_rad;
  b_rad=(M_PI * b_deg)/180.0;
  printf("pi/3 in Radians=%f\n", b_rad);

  
  //calculating area
  //f(x0)+f(xN)
  float area=tan(a)+tan(b_rad);
  printf("Initial sum=%f\n", area);


  //creating the loop
  //2*f(x1)+2f(x2)+ ...+ 2*f(xN-1)
  for(i=5; i<60; i=i+5) {
    area=area+2*tan((M_PI*i)/180.0);
    printf("New area at x%d =%f\n",  i/5, area);

  }
  
  //Multiply with b-a/2N
  float mult_rad=M_PI*((b_deg-a)/(2*N))/180.0;
  area=mult_rad*area;


  //Approximated value 
  printf("\nTrapezodial result=%f\n", area);
  //Actual result 
  printf("Real result=%f\n", log(2.0));                                     
  
  
return 0;
}
